### PR TITLE
[PEFT] minor fixes for consistencies

### PR DESCRIPTION
**Pull request by HollowMan6**
# What does this PR do ?

Some minor fixes

# Changelog

- Use `nn.Identity` when dropout == 0
- When lora_A_init_method == "xavier", it should use `xavier_uniform_` instead of `uniform_`
- `dropout_position` default to `pre` for consistency.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [X] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [X] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: src/megatron/bridge/peft/lora_layers.py [hunk 1]
```diff
@@ -162,11 +162,14 @@ def _init_adapter(
         obj.lora_a = nn.Linear(in_features, dim, bias=False, dtype=dtype, device=device)
         obj.lora_b = nn.Linear(dim, out_features, bias=False, dtype=dtype, device=device)
         if lora_A_init_method == "xavier":
-            torch.nn.init.uniform_(obj.lora_a.weight.data)
+            torch.nn.init.xavier_uniform_(obj.lora_a.weight.data)
```
[the remark this answers is not in the document]
Comment (HollowMan6) [reply src/megatron/bridge/peft/lora_layers.py]: fixed

File: src/megatron/bridge/peft/lora_layers.py [hunk 2]
```diff
@@ -546,11 +549,14 @@ def _init_adapter(
...
```
[the remark this answers is not in the document]
Comment (HollowMan6) [reply src/megatron/bridge/peft/lora_layers.py]: fixed

File: src/megatron/bridge/peft/utils.py [hunk 3]
```diff
@@ -406,7 +406,7 @@ def __init__(
         dropout: float = 0.0,
         model_parallel_config: Optional[ModelParallelConfig] = None,
         alpha: Optional[float] = None,
-        dropout_position: str = "post",
+        dropout_position: str = "pre",
```
[the remark this answers is not in the document]
Comment (HollowMan6) [reply src/megatron/bridge/peft/utils.py:409]: fixed

**Review by yaoyu-33 — DISMISSED** [review]

File: tests/unit_tests/peft/test_lora.py [hunk 4]
```diff
@@ -195,9 +195,10 @@ def test_lora_adapter_properties(self):
         # Check adapter properties
         adapter = transformed_model.linear_qkv
         assert hasattr(adapter, "dim")
+        assert hasattr(adapter, "alpha")
         assert hasattr(adapter, "scale")
-        assert hasattr(adapter, "lora_a")
-        assert hasattr(adapter, "lora_b")
+        assert hasattr(adapter, "linear_in")
+        assert hasattr(adapter, "linear_in")
```
[the remark this answers is not in the document]
Comment (HollowMan6) [reply tests/unit_tests/peft/test_lora.py]: fixed

File: tests/unit_tests/peft/test_lora.py [hunk 5]
```diff
@@ -494,13 +495,13 @@ def test_lora_reproducibility(self):
...
+        linear_in_1 = adapted_model1.linear_qkv.linear_in.weight.data
+        linear_in_2 = adapted_model2.linear_qkv.linear_in.weight.data
+        assert torch.equal(linear_in_1, linear_in_2)
 
-        lora_b_1 = adapted_model1.linear_qkv.lora_b.weight.data
-        lora_b_2 = adapted_model2.linear_qkv.lora_b.weight.data
-        assert torch.equal(lora_b_1, lora_b_2)
+        linear_in_1 = adapted_model1.linear_qkv.linear_out.weight.data
+        linear_in_2 = adapted_model2.linear_qkv.linear_out.weight.data
+        assert torch.equal(linear_in_1, linear_in_2)
```
[the remark this answers is not in the document]
Comment (HollowMan6) [reply tests/unit_tests/peft/test_lora.py]: fixed

**Review by yaoyu-33 — APPROVED** [review]
